source/tutorials/callpackage.md

```diff
@@ -0,0 +1,283 @@
+---
+date: 2022-09-08
+authors:
+  - Norbert Melzer
+  - Valentin Gagarin
+  - Matthias Meschede
+myst:
+  html_meta:
+    "keywords": "tutorial, callPackage, override, package, customise, parameters, nix, nixpkgs"
+---
+
+(callpackage-tutorial)=
+# Package parameters and overrides with `callPackage`
+
+Nix ships with a a special-purpose programming language for creating packages and configurations: the Nix language.
+It is used to build the Nix package collection, known as {term}`Nixpkgs`.
+
+Being purely functional, the Nix language allows declaring custom functions to abstract over common patterns.
+One of the most prominent patterns in Nixpkgs is parametrisation of package recipes.
+
+## Overview
+
+Nixpkgs is sizeable software project on it's own, with coding conventions and idioms that have emerged over the years.
+It has [established a convention](https://github.com/NixOS/nixpkgs/pull/9869) of composing parameterised packages with automatic settings through a function named [`callPackage`](https://github.com/NixOS/nixpkgs/commit/fd268b4852d39c18e604c584dd49a611dc795a9b).
```

Comment:
The PR there didn't establish the convention. Even before that PR, `callPackage` was used for the vast majority of packages, see [the file before the PR](https://github.com/Mathnerd314/nixpkgs/blob/c545bc80f68a44d0d0e1cc85290d74d1d57c09fa/pkgs/top-level/all-packages.nix).

Here's the actual commit that introduced `callPackage` as a convention: https://github.com/nixos/nixpkgs/commit/d17f0f9cbca38fabb71624f069cd4c0d6feace92

```suggestion
It has [established a convention](https://github.com/nixos/nixpkgs/commit/d17f0f9cbca38fabb71624f069cd4c0d6feace92) of composing parameterised packages with automatic settings through a function named [`callPackage`](https://github.com/NixOS/nixpkgs/commit/fd268b4852d39c18e604c584dd49a611dc795a9b).
```

Reply:
Uhh thanks for the dig-up!